### PR TITLE
feat(superuser): Require u2f on superuser access form

### PR DESCRIPTION
**Pull request by maxiuyuan**
As discussed in security Updates email, we will now require u2f on the superuser access form

### CODE REVIEW

File: static/app/components/modals/sudoModal.tsx [hunk 1]
```diff
@@ -42,36 +42,63 @@ type State = {
...
-      this.handleSuccess();
-    } catch (err) {
-      this.handleError(err);
+    });
+  };
+
+  handleSubmit = async data => {
+    const {api, isSuperuser} = this.props;
+    const superuserAccessCategory =
+      this.state.superuserAccessCategory === null
```
Comment (nhsiehgit) [inline static/app/components/modals/sudoModal.tsx:75]: can we just check falsey-ness?
```
const {superuserAccessCategory, superuserReason} = this.state;
const suAccessCategory = !superuserAccessCategory ? data.superuserAccessCategory : superuserReason;
```
?

File: static/app/components/modals/sudoModal.tsx [hunk 2]
```diff
@@ -42,36 +42,63 @@ type State = {
...
+        ? data.superuserAccessCategory
+        : this.state.superuserAccessCategory;
+
+    const superuserReason =
```
Comment (nhsiehgit) [inline static/app/components/modals/sudoModal.tsx:79]: const reason = superuserReason || data.superuserReason

File: static/app/components/modals/sudoModal.tsx (hunk 1 above)
Comment (nhsiehgit) [reply static/app/components/modals/sudoModal.tsx:75]: actually naw.
Lets just say `const suAccessCategory = this.state.superuserAccessCategory || data.superuserAccessCategory`

File: static/app/components/modals/sudoModal.tsx [hunk 3]
```diff
@@ -42,36 +42,63 @@ type State = {
...
+      this.state.superuserReason === null
+        ? data.superuserReason
+        : this.state.superuserReason;
+
+    if (!this.state.authenticators.length) {
```
Comment (nhsiehgit) [inline static/app/components/modals/sudoModal.tsx:84]: we can deconstruct the state at the top 
```
const { superuserAccessCategory, superuserReason, authenticators }  = this.state
```

and we can probably do this check at the top too so we return early

**Review by nhsiehgit — COMMENTED** [review]
hmmm...
a few clean up comments.

don't suppose we can get tests on this can we?

**Review by nhsiehgit — APPROVED** [review]
Nothing jumping out at me.

Not super familiar with this flow, but i'll give it a 👍 as long as its working

**Review by dashed — APPROVED** [review]

**Pull request merged**